frontend/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { createContext, useContext, useEffect, useRef, useState } from 'react';
+import { scheduleTokenRefresh, decodeToken } from '../../services/authService';
+import { useNavigate } from 'react-router-dom';
+import { message } from 'antd';
+
+type AuthContextType = {
+    isLoggedIn: boolean;
+    user: { [key: string]: any } | null;
+};
+
+const AuthContext = createContext<AuthContextType>({
+    isLoggedIn: false,
+    user: null,
+});
+
+export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+    const navigate = useNavigate();
+    const refreshTimeout = useRef<NodeJS.Timeout | null>(null);
+    const [isLoggedIn, setIsLoggedIn] = useState<boolean>(!!localStorage.getItem('access_token'));
+    const [user, setUser] = useState<{ [key: string]: any } | null>(null);
+
+    const decodeAndSetUser = (token: string | null) => {
+        if (!token) {
+            setUser(null);
+            return;
+        }
+
+        try {
+            const decoded = decodeToken(token);
+            setUser(decoded);
+            setIsLoggedIn(true);
+        } catch (error) {
+            console.error('Token decode error:', error);
+            setUser(null);
+        }
```

Comment:
It may be a good idea to show error message.

Reply:
And isn't that whats happening in 33 line?

Reply:
I meant on screen. Console is console. In useEffect you have `message.error('Log in again');`

Reply:
Done